### DISCUSSION TITLE
Infinite Canvas / AI features

### DISCUSSION THREAD

**Discussion by vasnt** [marimo-team/marimo#1914] [Ideas]
- Possible to have Infinite Canvas/2D mode ?
- Possible it add some AI features like autofix errors? (borrowed from :https://github.com/pretzelai/pretzelai)
- Possible to have a hybrid (Note-taking+Code-editing/running) experience e.g. Obsidian/Note-taking + Code-editing/VscCode/Notebook

**Comment by akshayka** [comment]
Thanks for the questions!

> Possible to have Infinite Canvas/2D mode ?

We have a grid layout for presentation but not for editing. We likely won't get to infinite canvas / 2D mode soon, though it is a cool idea.

https://docs.marimo.io/guides/apps.html#grid-layout

> Possible it add some AI features like autofix errors? 

We have AI features already, including generate with AI and autofix, but you need to enable them first:

<img width="616" alt="image" src="https://github.com/user-attachments/assets/6ea25b99-a497-423d-9d3b-bd4947d46894">

<img width="340" alt="image" src="https://github.com/user-attachments/assets/71f90087-9168-4d16-accc-f0ad338504c2">

You can experiment with them for free at https://marimo.new

> Possible to have a hybrid (Note-taking+Code-editing/running) experience e.g. Obsidian/Note-taking + Code-editing/VscCode/Notebook

We support saving marimo notebooks as markdown. For example, you can run this notebook with `marimo edit`, even though the extension is `.md`: https://github.com/marimo-team/marimo/blob/main/marimo/_tutorials/markdown_format.md
